src/msw/ole/oleutils.cpp

```diff
@@ -71,27 +71,63 @@ WXDLLEXPORT wxString wxConvertStringFromOle(BSTR bStr)
 // wxBasicString
 // ----------------------------------------------------------------------------
 
-wxBasicString::wxBasicString(const wxString& str)
+wxBasicString::wxBasicString(BSTR bstr)
 {
-    m_bstrBuf = SysAllocString(str.wc_str(*wxConvCurrent));
+	m_bstrBuf = bstr;
 }
 
-wxBasicString::wxBasicString(const wxBasicString& src)
+wxBasicString::wxBasicString(const wxString& str)
 {
-    m_bstrBuf = src.Get();
+    m_bstrBuf = SysAllocString(str.wc_str(*wxConvCurrent));
 }
 
-wxBasicString& wxBasicString::operator=(const wxBasicString& src)
+wxBasicString::wxBasicString(const wxBasicString& bstr)
 {
-    SysReAllocString(&m_bstrBuf, src);
-    return *this;
+    m_bstrBuf = bstr.Copy();
 }
 
 wxBasicString::~wxBasicString()
 {
     SysFreeString(m_bstrBuf);
 }
 
+wxBasicString& wxBasicString::operator=(const wxBasicString& src)
+{
+	if ( this != &src )
+	{
+		SysFreeString(m_bstrBuf);
+		m_bstrBuf = src.Copy();
+	}
+
+	return *this;
+}
+
+wxBasicString& wxBasicString::operator=(const wxString& str)
+{
+	SysFreeString(m_bstrBuf);
+	m_bstrBuf = ::SysAllocString(str.wc_str(*wxConvCurrent));
+
+	return *this;
+}
+
+wxBasicString& wxBasicString::operator=(BSTR bstr)
+{
+	wxCHECK_MSG(m_bstrBuf != bstr, *this, wxS("Attempting to attach already attached BSTR!"));

```

Comment:
I realized that this check does not account for a valid situation where both `m_bstrBuf` and `bstr` are NULL. The condition should be rewritten as `m_bstrBuf == NULL || m_bstrBuf != bstr`.